mongosql/src/schema/definitions.rs

```diff
@@ -999,10 +1014,8 @@ impl Schema {
                         _ => Either::Right(s),
                     });
                 if !docs.is_empty() {
-                    let doc_schema = Schema::Document(
-                        docs.into_iter()
-                            .fold(Document::default(), |acc, s| acc.merge(s)),
-                    );
+                    let doc_schema =
+                        Schema::Document(docs.into_iter().reduce(|acc, s| acc.union(s)).unwrap());
```

Comment:
This is the core change this ticket was concerned with! Funny how big the ramifications were.